tests/config/commonFixtures.ts

```diff
@@ -32,28 +33,59 @@ import childProcess from 'child_process';
 
 type ProcessData = {
   pid: number, // process ID
-  pgid: number, // process groupd ID
+  pgrp: number, // process groupd ID
   children: Set<ProcessData>, // direct children of the process
 };
 
-function buildProcessTreePosix(pid: number): ProcessData {
+function readAllProcessesLinux(): { pid: number, ppid: number, pgrp: number }[] {
+  const result: {pid: number, ppid: number, pgrp: number}[] = [];
+  for (const dir of fs.readdirSync('/proc')) {
+    const pid = +dir;
+    if (isNaN(pid))
+      continue;
+    try {
+      const statFile = fs.readFileSync(`/proc/${pid}/stat`, 'utf8');
+      // Format of /proc/*/stat is described https://man7.org/linux/man-pages/man5/proc.5.html
+      const match = statFile.match(/^(?<pid>\d+)\s+\((?<comm>.*)\)\s+(?<state>R|S|D|Z|T|t|W|X|x|K|W|P)\s+(?<ppid>\d+)\s+(?<pgrp>\d+)/);
+      if (match) {
+        result.push({
+          pid: +match.groups.pid,
+          ppid: +match.groups.ppid,
+          pgrp: +match.groups.pgrp,
+        });
+      }
+    } catch (e) {
+      // We don't have access to some /proc/<pid>/stat file.
```

Comment:
should we log this for easier debugging?

Reply:
I don't think we should: all the descendants should be accessible, and all the other processes (e.g. root-spawned) might happen regularly. This is not an edge-case.